R/lint.R

```diff
@@ -114,14 +117,7 @@ lint <- function(filename, linters = NULL, ..., cache = FALSE, parse_settings =
     }
   }
 
-  if (inherits(source_expressions$error, "lint")) {
-    lints[[itr <- itr + 1L]] <- source_expressions$error
-
-    if (isTRUE(cache)) {
-      cache_lint(lint_cache, list(filename = filename, content = ""), "error", source_expressions$error)
-    }
-  }
-
+  lints <- maybe_append_error_lint(lints, source_expressions$error, lint_cache, filename)
```

Comment:
could you double-check for any performance implications here? this'll copy the list whenever an edit is needed

Reply:
Will do once refactoring is complete.